jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -843,28 +844,63 @@ ecma_builtin_array_prototype_object_slice (ecma_value_t arg1, /**< start */
 
   ecma_object_t *new_array_p = ecma_get_object_from_value (new_array);
 
+  if (copied_length == 0)
+  {
+#if ENABLED (JERRY_ES2015)
+    ecma_value_t set_length_value = ecma_builtin_array_prototype_helper_set_length (new_array_p, ((ecma_number_t) 0));
```

Comment:
Hmm, I didn't realize we'd have to set the length, my bad. I'd rather not duplicate this, could you please remove the early return and do `if (use_fast_path && copied_length > 0)`? That should have the same result.